src/transaction/transaction.rs

```diff
@@ -374,6 +392,9 @@ impl Transaction {
     /// # });
     /// ```
     pub async fn lock_keys(&self, keys: impl IntoIterator<Item = impl Into<Key>>) -> Result<()> {
+        if self.is_pessimistic() {
```

Comment:
Because in pessimistic transactions we should use `get_for_update` to pessimistically lock the key, which also writes a `Lock` record in the buffer.

Reply:
Does it make sense for the user to want to lock a key without getting the value? If so, could we insert a lock into the buffer in this method?

Reply:
Yeah we can do this. 
So `lock_keys` will have different semantics in optimistic and pessimistic txns. Then it will be quite similar to `batch_get_for_update` in pessimistic txns, except that it doesn't return values.